lib/adapters/idb/index.js

```diff
@@ -480,7 +480,16 @@ function init(api, opts, callback) {
 
         var change = opts.processChange(winningDoc, metadata, opts);
         change.seq = metadata.seq;
-        if (filter(change)) {
+
+        var filtered;
+        try {
+          filtered = filter(change);
+        } catch(err) {
+          var msg = 'Filter function threw: ' + err.toString();
+          return opts.complete(errors.error(errors.BAD_REQUEST, msg));
+        }
+
+        if (filtered) {
```

Comment:
Normally `try/catch`es should always go into their own functions to avoid deoptimization of the entire function. Normally I wouldn't nitpick, but I do know that `onGetWinningDoc()` is called a lot.